File: interface/Go/NetWork.go
//v0.1.1
package mlg

import (
	"io/ioutil"
	"net/http"
	"net/url"
	"strings"
	"time"
)

const ONLY4NETERR int = 0
const NETERRorTO int = 1

func pureGet(seg []string) string {
	str2get := "http://127.0.0.1:" + strings.Join(seg, "/")
	resp, err := http.Get(str2get)
	if err!=nil {
		//fmt.Println(err)
		return "er86756"
	}
	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	//fmt.Println(string(body))
	if err!=nil{
		return "er56578"
	}
	return "ok" + string(body)
}

func purePost(seg []string, data map[string]string) string {
	str2post := "http://127.0.0.1:" + strings.Join(seg, "/")
	urlvalues := url.Values{}
	for k := range data {
		urlvalues.Add(k, data[k])
	}
	resp, err := http.PostForm(str2post, urlvalues)
	if err!= nil {
		return "er45347"
	}
	body, err := ioutil.ReadAll(resp.Body)
	if err!= nil {
		return "er877587"
	}
	return "ok"+string(body)
}


func loopPost(seg []string, data map[string]string ,loopFor int) string {
	loopFlag := true
	get := "er68564"
	for loopFlag {
		get = purePost(seg, data)
		if loopFor == ONLY4NETERR{
			if get[0:2] != "er"{
				loopFlag = false
			}

		} else {
			if get[0:2] == "er" || get[0:4]=="okto"{
				loopFlag = true
			} else {
				loopFlag = false
			}
		}
		time.Sleep(50*time.Millisecond)
	}
	return get
}

func loopGet(seg []string, loopFor int) string {
	loopFlag := true
	get := "er684876"
	for loopFlag {
		get = pureGet(seg)
		if loopFor == ONLY4NETERR{
			if get[0:2] != "er"{
				loopFlag = false
			}

		} else {
			if get[0:2] == "er" || get[0:4]=="okto"{
				loopFlag = true
			} else {
				loopFlag = false
			}
		}
		time.Sleep(50*time.Millisecond)
	}
	return get
}

File: interface/Python/NetWork.py
import requests
import time
# v0.1.1

class NetWork:
    ONLY4NETERR = 0
    NETERRorTO = 1

    # @classmethod
    # def getTool(cls, loop, *seg):
    #     str2get = 'http://127.0.0.1:' + "/".join(seg)
    #     if loop:
    #         while True:
    #             try:
    #                 get = requests.get(str2get)
    #                 return get
    #             except Exception:
    #                 print("我也不知道什么错误")
    #             time.sleep(0.05)
    #     else:
    #         try:
    #             get = requests.get(str2get)
    #             err = None
    #         except ConnectionError:
    #             print('connection error, 暂时怀疑是服务程序未启动')
    #             get = None
    #             err = "err7345843"
    #         return get, err
    #
    # @classmethod
    # def postTool(cls, loop, *seg, **data):
    #     str2post = 'http://127.0.0.1:' + "/".join(seg)
    #     if loop:
    #         while True:
    #             try:
    #                 get = requests.post(str2post, data)
    #                 return get
    #             except Exception:
    #                 print(str2post)
    #                 print("一个错误")
    #                 time.sleep(0.05)
    #                 raise
    #
    #     else:
    #         try:
    #             get = requests.post(str2post, data)
    #             err = None
    #         except ConnectionError:
    #             print("connection err yiwaichucuo")
    #             get = None
    #             err = "err87809"
    #         return get, err

    @classmethod
    def pureGet(cls, seg) -> str:
        str2get = 'http://127.0.0.1:' + "/".join(seg)
        try:
            get = requests.get(str2get)
            return "ok" + get.text
        except ConnectionError:
            return "er87098"
        except Exception:
            return "er9887980"

    @classmethod
    def purePost(cls, seg, data) -> str:
        str2post = 'http://127.0.0.1:' + "/".join(seg)
        try:
            get = requests.post(str2post, data)
            return "ok" + get.text
        except ConnectionError:
            return "er76987"
        except Exception:
            return "er9834"

    @classmethod
    def loopPost(cls, seg, data, loopFor) -> str:
        loopFlag = True
        get = "er5667"
        while loopFlag:
            get = NetWork.purePost(seg, data)
            if loopFor == NetWork.ONLY4NETERR:
                if get[:2] != "er":
                    loopFlag = False
            else:
                if get[:2] == "er" or get[:4] == "okto":
                    loopFlag = True
                else:
                    loopFlag = False
            time.sleep(0.05)
        return get

    @classmethod
    def loopGet(cls, seg, loopFor) -> str:
        loopFlag = True
        get = "er98096"
        while loopFlag:
            get = NetWork.pureGet(seg)
            if loopFor == NetWork.ONLY4NETERR:
                if get[:2] != "er":
                    loopFlag = False
            else:
                if get[:2] == "er" or get[:4] == "okto":
                    loopFlag = True
                else:
                    loopFlag = False
            time.sleep(0.05)
        return get
